src/amcl-extensions/big_256_56.c

```diff
@@ -84,12 +87,15 @@ void big_256_56_mod_mul_and_add(BIG_256_56 *big_out,
 }
 
 void big_256_56_random_mod_order(BIG_256_56 *big_out,
-                                 csprng *rng)
+                                 struct ecdaa_prng *prng)
 {
+    if (ECDAA_PRNG_INITIALIZED_YES != prng->initialized)
+        abort();
+
     BIG_256_56 curve_order;
     BIG_256_56_rcopy(curve_order, CURVE_Order_BN254);
 
-    BIG_256_56_randomnum(*big_out, curve_order, rng);
+    BIG_256_56_randomnum(*big_out, curve_order, &prng->impl);
```

Comment:
Continuing the prior comment, replace this with
```
BIG_256_56_randomnum(*big_out, curve_order, ecdaa_prng_get_csprng(prng));
```
where `ecdaa_prng_get_csprng(...)` returns `csprng*` or calls `abort()`.

Reply:
I made _all_ the functions take `ecdaa_prng` rather than letting the `amcl-extensions` take the bare one, to make it easier to see where the bare one is getting used. 

If I move to the accessor discussed above, that would mean a grep of the source code would immediately show us where the bare csprng is being used directly.

Reply:
Just grep on everything except `acml-extensions` (or throw a ` | grep -v acml-extensions` on the end of your shell command to filter out those matches).

It depends on what is meant by `acml-extensions`. In my mind, those are things that would ideally be provided by the relevant "classes" in `acml` (and might sometime be contributed back).  So to the main `ecdaa` code, functions from `acml` or `acml-extensions` should be indistinguishable.

Reply:
Hmm, I hadn't been thinking of `amcl-extensions` that way. I'm sure I understand the advantage of thinking of it that way. Is that what others would expect from such a sub-directory?

There's only one function in `amcl-extensions` that uses the `csprng`, and thus really only one place in our code where we _actually_ use an rng: `big_256_56_random_mod_order`. The implementation of that function may change (cf. #1 ), so maybe that should get lifted out of `amcl-extensions`?

Reply:
I guess I'm a little confused as the to the point of `big256_56_random_mod_order()` in the first place.  Why not just call `BIG_256_56_randomnum(*big_out, CURVE_Order_BN254, rng)` directly?  I don't see the purpose of the `BIG_256_56_rcopy` call.

Reply:
As always, after thinking about it, I think I see your point. It's nice to not have the cognitive friction of having two conventions when using AMCL-related code in our project. Besides, if we end up wanting to submit any of our extensions to the AMCL project for inclusion, this makes it easier.

Reply:
Hmm, weird, I didn't see your question about `rcopy` when I left my previous comment.

That's copying from AMCL's examples. Concretely, it has to do with `const` vs non-`const`. The `r` seems to indicate `ROM` in their naming scheme. It does look like simply casting away const might work, but the fact that they explicitly do the copy in their examples makes me wonder if they're trying to be careful there (cf. my worries in #4 ).

So, at the moment, it's simply a wrapper to do the copy from "ROM" then call `_randomnum`, but I don't feel confident removing it.